test/units/module_utils/common/text/test_converters.py

```diff
@@ -0,0 +1,240 @@
+
+        It must fail with TypeError.
+        """
+        with pytest.raises(TypeError):
+            _json_encode_fallback(test_input)
+
+
+class Test_container_to_bytes:
+
+    """Namespace for testing container_to_bytes()."""
+
+    @pytest.mark.parametrize(
+        'test_input,expected',
+        [
+            ({1: 1}, {1: 1}),
+            ([1, 2], [1, 2]),
+            ((1, 2), (1, 2)),
+            (1, 1),
+            (1.1, 1.1),
+            ('str', b'str'),
+            (['str'], [b'str']),
+            (('str'), (b'str')),
+            ({'str': 'str'}, {b'str': b'str'}),
+        ]
+    )
+    @pytest.mark.parametrize('encoding', ['utf-8', 'ascii'])
+    @pytest.mark.parametrize('errors', ['strict', 'ignore'])
+    def test_to_bytes_different_types(self, test_input, expected, encoding, errors):
+        """Test for passing objects to container_to_bytes()."""
+
+        assert container_to_bytes(test_input, encoding=encoding, errors=errors) == expected
+
+    @pytest.mark.parametrize(
+        'test_input,expected',
+        [
+            ({1: 1}, {1: 1}),
+            ([1, 2], [1, 2]),
+            ((1, 2), (1, 2)),
+            (1, 1),
+            (1.1, 1.1),
+            (True, True),
+            (None, None),
+            ('str', 'str'.encode(DEFAULT_ENCODING)),
+            (['str'], ['str'.encode(DEFAULT_ENCODING)]),
+            (('str'), ('str'.encode(DEFAULT_ENCODING))),
+            ({'str': 'str'}, {'str'.encode(DEFAULT_ENCODING): 'str'.encode(DEFAULT_ENCODING)}),
+        ]
+    )
+    def test_to_bytes_default_encoding_err(self, test_input, expected):
+        """
+        Test for passing objects to container_to_bytes(). Default encoding and errors
+        """
+
+        assert container_to_bytes(test_input) == expected
+
+
+class Test_container_to_text:
+
+    """Namespace for testing container_to_text()."""
+
+    @pytest.mark.parametrize(
+        'test_input,expected',
+        [
+            ({1: 1}, {1: 1}),
+            ([1, 2], [1, 2]),
+            ((1, 2), (1, 2)),
+            (1, 1),
+            (1.1, 1.1),
+            (b'str', 'str'),
+            ([b'str'], ['str']),
+            ((b'str'), ('str')),
+            ({b'str': b'str'}, {'str': 'str'}),
+        ]
+    )
+    @pytest.mark.parametrize('encoding', ['utf-8', 'ascii'])
+    @pytest.mark.parametrize('errors', ['strict', 'ignore'])
+    def test_to_text_different_types(self, test_input, expected, encoding, errors):
+        """Test for passing objects to container_to_text()."""
+
+        assert container_to_text(test_input, encoding=encoding, errors=errors) == expected
+
+    @pytest.mark.parametrize(
+        'test_input,expected',
+        [
+            ({1: 1}, {1: 1}),
+            ([1, 2], [1, 2]),
+            ((1, 2), (1, 2)),
+            (1, 1),
+            (1.1, 1.1),
+            (True, True),
+            (None, None),
+            ('str'.encode(DEFAULT_ENCODING), 'str'),
```

Comment:
Why are you testing native strings only? Where's unicode strings?

Reply:
good note, I'll remember, thanks!